crates/agglayer-types/src/lib.rs

```diff
@@ -219,6 +228,40 @@ impl Certificate {
         ])
         .into()
     }
+
+    /// Returns the L1 Info Tree leaf count considered for this [`Certificate`].
+    /// Corresponds to the highest L1 Info Tree leaf index considered by the
+    /// imported bridge exits.
+    pub fn l1_info_tree_leaf_count(&self) -> u32 {
+        self.imported_bridge_exits
+            .iter()
+            .map(|i| i.l1_leaf_index())
+            .max()
+            .unwrap_or(Self::DEFAULT_L1_INFO_LEAF_COUNT)
+    }
+
+    /// Returns the L1 Info Root considered for this [`Certificate`].
+    /// Fails if multiple L1 Info Root are considered among the inclusion proofs
+    /// of the imported bridge exits.
+    pub fn l1_info_root(&self) -> Result<Digest, Error> {
+        let Some(l1_info_root) = self
+            .imported_bridge_exits
+            .first()
```

Comment:
Why do we take the first one?

Reply:
just ensuring that they are all equals, so just picking the first one and checking that all others are equal. If there is none, then returning the default hash defined by the constant